### ISSUE TITLE
Adding Twilight Forest as a dimension for Frozen Apocalypse to take affect in causes some rather unusual buggy behavior in said dimension.

### ISSUE THREAD

**Issue by FrozenDragon0** [Jaggwagg/frozen-apocalypse#12]
**Describe the bug**
Essentially when Twilight Forest is added as a dimension for FA to take affect in the dimension experiences some very buggy behavior when generating new chunks.

**To Reproduce**
1. Add Twilight Forest as a dimension in the FA's config file.
2. Generate a new world and set the apocalypse level to 9.
3. Make a Twilight Forest portal and explore new chunks.
4. Observe and notice that any new chunks generated in the Twilight Forest are half covered in snow with water frozen over, yet none of FA's actual effects seem to take place in said dimension.

**Expected behavior**
Frozen Apocalypse to work correctly in said dimension like it does in the Overworld.

**Screenshots**
![2024-01-13_08 43 05](https://github.com/Jaggwagg/frozen-apocalypse-mod/assets/18079316/98314566-5fbb-47de-a894-a64a7737d714)

**Additional context**
Has the mod been updated to freeze over any newly generated chunks that haven't been touched before? As I've noticed new chunks in the Overworld are already frozen over.

**Comment by Jaggwagg** [comment]
Oh yeah, that's a feature! It's suppose to already be frozen as you have reached the stage where it freezes and snows. Unless you haven't (should be day 3).

As for the effects not working, does it say day 3 your F3 menu?

edit: Yeah, it's working as intended. You set the apocalypse level to 9. Level 3 and after will cause biomes to snow. Snowy biomes will generate with ice and snow as intended. However, I would like to know if the effects (like leaves and grass changing/decaying) are not working. That may be a Forgified Fabric issue with mods and custom blocks.

**Comment by FrozenDragon0** [comment]
> Oh yeah, that's a feature! It's suppose to already be frozen as you have reached the stage where it freezes and snows. Unless you haven't (should be day 3).
> 
> As for the effects not working, does it say day 3 your F3 menu?
> 
> edit: Yeah, it's working as intended. You set the apocalypse level to 9. Level 3 and after will cause biomes to snow. Snowy biomes will generate with ice and snow as intended. However, I would like to know if the effects (like leaves and grass changing/decaying) are not working. That may be a Forgified Fabric issue with mods and custom blocks.

The grass changing and leaves decaying effect does not work in the dimension, but I did grow one of the mod's unique trees in the overworld and the leaves did decay as expected albeit in a 3x3 pattern strangely, entities freezing also did not work in the dimension either.

Regarding new chunks being frozen over after day 3 that's a neat feature I was not expecting! :)

**Comment by Jaggwagg** [comment]
Alright, there was a problem with the dimension id checking. However, it shouldn't have placed the snow anyway. I tested it using the latest version of Twilight Forest before the fix, and it wasn't even placing the snow.

So, I will consider the bug to be fixed. Feel free to reopen the issue once 1.2.1 comes out if it is still behaving weirdly. Might be a mod conflict somewhere.

edit: I tested it with no other mods except for Twilight for reference.

**Comment by Jaggwagg** [comment]
Nevermind, that didn't fix it.

**Comment by FrozenDragon0** [comment]
> Nevermind, that didn't fix it.

After doing some further testing, it would seem Frozen Apocalypse placing snow in new chunks completely ignores the dimension setting and is still doing so for Twilight Forest despite the fact I've not listed it as a dimension for FA.

From what I understand to trigger this is you need to have the apocalypse level set to phase 3 or after, wait for a bit in the overworld and then make a Twilight Forest portal and head through it, you'll then see new generated chunks are snow covered.
This is what I believe triggers it, still playing around with it however.

**Comment by Jaggwagg** [comment]
Tested the slight change to the previous fix and it works as expected, even with Twilight Forest and other mods.

**Comment by FrozenDragon0** [comment]
I've updated Frozen Apocalypse to 1.2.1 and it appears that snow is still being placed in new chunks in Twilight Forest even though I don't have the dimension listed as a dimension the mod can take affect in, and I've separated this from other mods now so it's just Twilight Forest and Forgified Fabric + Frozen Apocalypse, so definitely not a mod conflict.

The placeSnow setting in Frozen Apocalypse's config seems to be failing in ignoring Twilight Forest outright.
![2024-01-20_08 28 12](https://github.com/Jaggwagg/frozen-apocalypse-mod/assets/18079316/edb84470-ce13-4c43-bea6-34fe1af18939)

If you want me to make a separate issue regarding this I'm happy to.
